### ISSUE TITLE
Build action for aqt.exe failed: gravitybee raise exception in hook-aqt

### ISSUE THREAD

**Issue by miurahr** [miurahr/aqtinstall#556]
**Describe the bug**
Release actions script that run gravitybee failed 

https://github.com/miurahr/aqtinstall/runs/7807796241?check_suite_focus=true

**To Reproduce**

The failure happened from https://github.com/miurahr/aqtinstall/commit/d30618791c0198bf619a7874991495a589353cdb

**Expected behavior**

succeeded to produce aqt.exe

**build log**

```
Requirement already satisfied: click in d:\a\aqtinstall\aqtinstall\build_venv\lib\site-packages (from gravitybee) (8.1.3)
Requirement already satisfied: colorama in d:\a\aqtinstall\aqtinstall\build_venv\lib\site-packages (from click->gravitybee) (0.4.5)
Collecting altgraph
  Downloading altgraph-0.17.2-py2.py3-none-any.whl (21 kB)
Collecting pywin32-ctypes>=0.2.0
  Downloading pywin32_ctypes-0.2.0-py2.py3-none-any.whl (28 kB)
Requirement already satisfied: setuptools in d:\a\aqtinstall\aqtinstall\build_venv\lib\site-packages (from PyInstaller->gravitybee) (63.4.1)
Collecting pyinstaller-hooks-contrib>=2021.4
  Downloading pyinstaller_hooks_contrib-2022.8-py2.py3-none-any.whl (239 kB)
     ------------------------------------- 239.8/239.8 kB 15.3 MB/s eta 0:00:00
Collecting pefile>=2022.5.30
  Downloading pefile-2022.5.30.tar.gz (72 kB)
     ---------------------------------------- 72.9/72.9 kB 3.9 MB/s eta 0:00:00
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting future
  Downloading future-0.18.2.tar.gz (829 kB)
     ------------------------------------- 829.2/829.2 kB 26.4 MB/s eta 0:00:00
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Building wheels for collected packages: pefile, future
  Building wheel for pefile (setup.py): started
  Building wheel for pefile (setup.py): finished with status 'done'
  Created wheel for pefile: filename=pefile-2022.5.30-py3-none-any.whl size=69362 sha256=3c0a0c35819947733dcad572143ff535a8204b7e05781cdc8abc46f862785a01
  Stored in directory: c:\users\runneradmin\appdata\local\pip\cache\wheels\f1\fe\01\5536332fc49e010786c52886ac14f72073dd65fed3c3d8945c
  Building wheel for future (setup.py): started
  Building wheel for future (setup.py): finished with status 'done'
  Created wheel for future: filename=future-0.18.2-py3-none-any.whl size=491058 sha256=543ea3c5491cf0b[233](https://github.com/miurahr/aqtinstall/runs/7807796241?check_suite_focus=true#step:7:234)a19845f034bf9704740327f94befca177b8905a7e2681b
  Stored in directory: c:\users\runneradmin\appdata\local\pip\cache\wheels\2f\a0\d3\4030d9f80e6b3be787f19fc911b8e7aa462986a40ab1e4bb94
Successfully built pefile future
Installing collected packages: pywin32-ctypes, altgraph, pyinstaller-hooks-contrib, future, pyppyn, pefile, PyInstaller, gravitybee
Successfully installed PyInstaller-5.3 altgraph-0.17.2 future-0.18.2 gravitybee-0.4.0 pefile-2022.5.30 pyinstaller-hooks-contrib-2022.8 pyppyn-0.5.0 pywin32-ctypes-0.2.0
D:\a\aqtinstall\aqtinstall\build_venv\lib\site-packages\setuptools\installer.py:27: SetuptoolsDeprecationWarning: setuptools.installer is deprecated. Requirements should be satisfied by a PEP 517 installer.
  warnings.warn(
d:\a\aqtinstall\aqtinstall\.eggs\setuptools_scm-7.0.5-py3.9.egg\setuptools_scm\git.py:135: UserWarning: "D:\a\aqtinstall\aqtinstall" is shallow and may cause errors
  warnings.warn(f'"{wd.path}" is shallow and may cause errors')
D:\a\aqtinstall\aqtinstall\build_venv\lib\site-packages\setuptools\config\_apply_pyprojecttoml.py:196: UserWarning: `install_requires` overwritten in `pyproject.toml` (dependencies)
  warnings.warn(msg)
running bdist_wheel
running build
running build_py
creating build
creating build\lib
creating build\lib\aqt
copying aqt\archives.py -> build\lib\aqt
copying aqt\exceptions.py -> build\lib\aqt
copying aqt\helper.py -> build\lib\aqt
copying aqt\installer.py -> build\lib\aqt
copying aqt\metadata.py -> build\lib\aqt
copying aqt\updater.py -> build\lib\aqt
copying aqt\version.py -> build\lib\aqt
copying aqt\__init__.py -> build\lib\aqt
copying aqt\__main__.py -> build\lib\aqt
running egg_info
writing aqtinstall.egg-info\PKG-INFO
warning: no files found matching '*.txt'
writing dependency_links to aqtinstall.egg-info\dependency_links.txt
writing entry points to aqtinstall.egg-info\entry_points.txt
writing requirements to aqtinstall.egg-info\requires.txt
writing top-level names to aqtinstall.egg-info\top_level.txt
reading manifest template 'MANIFEST.in'
warning: no previously-included files found matching 'qtaccount.ini'
warning: no previously-included files found matching 'aqtinstall.log'
adding license file 'LICENSE'
D:\a\aqtinstall\aqtinstall\build_venv\lib\site-packages\setuptools\command\install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
writing manifest file 'aqtinstall.egg-info\SOURCES.txt'
copying aqt\combinations.json -> build\lib\aqt
copying aqt\logging.ini -> build\lib\aqt
copying aqt\settings.ini -> build\lib\aqt
Traceback (most recent call last):
  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\runpy.py", line 197, in _run_module_as_main
installing to .\.pyppyn\temp
running install
running install_lib
creating .pyppyn\temp
creating .pyppyn\temp\aqt
copying build\lib\aqt\archives.py -> .\.pyppyn\temp\.\aqt
copying build\lib\aqt\combinations.json -> .\.pyppyn\temp\.\aqt
copying build\lib\aqt\exceptions.py -> .\.pyppyn\temp\.\aqt
copying build\lib\aqt\helper.py -> .\.pyppyn\temp\.\aqt
copying build\lib\aqt\installer.py -> .\.pyppyn\temp\.\aqt
copying build\lib\aqt\logging.ini -> .\.pyppyn\temp\.\aqt
copying build\lib\aqt\metadata.py -> .\.pyppyn\temp\.\aqt
copying build\lib\aqt\settings.ini -> .\.pyppyn\temp\.\aqt
copying build\lib\aqt\updater.py -> .\.pyppyn\temp\.\aqt
copying build\lib\aqt\version.py -> .\.pyppyn\temp\.\aqt
copying build\lib\aqt\__init__.py -> .\.pyppyn\temp\.\aqt
copying build\lib\aqt\__main__.py -> .\.pyppyn\temp\.\aqt
running install_egg_info
Copying aqtinstall.egg-info to .\.pyppyn\temp\.\aqtinstall-2.2.2-py3.9.egg-info
running install_scripts
adding license file "LICENSE" (matched pattern "LICEN[CS]E*")
creating .\.pyppyn\temp\aqtinstall-2.2.2.dist-info\WHEEL
creating '.\.pyppyn\dist\aqtinstall-2.2.2-py2.py3-none-any.whl' and adding '.\.pyppyn\temp' to it
adding 'aqt/__init__.py'
adding 'aqt/__main__.py'
adding 'aqt/archives.py'
adding 'aqt/combinations.json'
adding 'aqt/exceptions.py'
adding 'aqt/helper.py'
adding 'aqt/installer.py'
adding 'aqt/logging.ini'
adding 'aqt/metadata.py'
adding 'aqt/settings.ini'
adding 'aqt/updater.py'
adding 'aqt/version.py'
adding 'aqtinstall-2.2.2.dist-info/LICENSE'
adding 'aqtinstall-2.2.2.dist-info/METADATA'
adding 'aqtinstall-2.2.2.dist-info/WHEEL'
adding 'aqtinstall-2.2.2.dist-info/entry_points.txt'
adding 'aqtinstall-2.2.2.dist-info/top_level.txt'
adding 'aqtinstall-2.2.2.dist-info/RECORD'
removing .\.pyppyn\temp
GravityBee CLI, 0.4.0
2022-08-12 14:14:30,503 - gravitybee - INFO - Arguments:
2022-08-12 14:14:30,503 - gravitybee - INFO - app_name: aqtinstall
2022-08-12 14:14:30,503 - gravitybee - INFO - app_version: 2.2.2
2022-08-12 14:14:30,503 - gravitybee - INFO - operating_system: windows
2022-08-12 14:14:30,503 - gravitybee - INFO - machine_type: amd64
2022-08-12 14:14:30,503 - gravitybee - INFO - console_script: aqt
2022-08-12 14:14:30,503 - gravitybee - INFO - pkg_name: aqt
2022-08-12 14:14:30,503 - gravitybee - INFO - script_path: None
2022-08-12 14:14:30,503 - gravitybee - INFO - pkg_dir: .
2022-08-12 14:14:30,503 - gravitybee - INFO - src_dir: .
2022-08-12 14:14:30,503 - gravitybee - INFO - name_format: {an}-{v}-standalone-{os}-{m}
2022-08-12 14:14:30,503 - gravitybee - INFO - clean: False
2022-08-12 14:14:30,503 - gravitybee - INFO - work_dir: .gravitybee\build\136ffc041a4911ed
2022-08-12 14:14:30,503 - gravitybee - INFO - onedir: False
2022-08-12 14:14:30,503 - gravitybee - INFO - include_setup_extras: False
2022-08-12 14:14:30,503 - gravitybee - INFO - staging_dir: .gravitybee\dist
2022-08-12 14:14:30,503 - gravitybee - INFO - with_latest: True
2022-08-12 14:14:30,503 - gravitybee - INFO - sha: info
2022-08-12 14:14:30,503 - gravitybee - INFO - hook_template: D:\a\aqtinstall\aqtinstall\build_venv\lib\site-packages\gravitybee\hook-template
2022-08-12 14:14:30,503 - gravitybee - INFO - Package generator:
2022-08-12 14:14:30,503 - gravitybee - INFO - standalone_name: aqtinstall-2.2.2-standalone-windows-amd64
2022-08-12 14:14:30,503 - gravitybee - INFO - Created hook file: .gravitybee\build\136ffc041a4911ed\hooks\hook-aqt.py
    return _run_code(code, main_globals, None,
  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "D:\a\aqtinstall\aqtinstall\build_venv\Scripts\gravitybee.exe\__main__.py", line 7, in <module>
  File "D:\a\aqtinstall\aqtinstall\build_venv\lib\site-packages\click\core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "D:\a\aqtinstall\aqtinstall\build_venv\lib\site-packages\click\core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "D:\a\aqtinstall\aqtinstall\build_venv\lib\site-packages\click\core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "D:\a\aqtinstall\aqtinstall\build_venv\lib\site-packages\click\core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "D:\a\aqtinstall\aqtinstall\build_venv\lib\site-packages\gravitybee\cli.py", line 212, in main
    sys.exit(package_generator.generate())
  File "D:\a\aqtinstall\aqtinstall\build_venv\lib\site-packages\gravitybee\__init__.py", line 854, in generate
    shutil.copy2(self.args.info["script_path"], self._temp_script)
  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\shutil.py", line 444, in copy2
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\shutil.py", line [243](https://github.com/miurahr/aqtinstall/runs/7807796241?check_suite_focus=true#step:7:244), in copyfile
    if _samefile(src, dst):
  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\shutil.py", line 220, in _samefile
    return os.path.samefile(src, dst)
  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\genericpath.py", line 100, in samefile
    s1 = os.stat(f1)
TypeError: stat: path should be string, bytes, os.PathLike or integer, not NoneType
The system cannot find the path specified.
Error: Process completed with exit code 1.
```

**Additional context**

GravityBee project related issue: https://github.com/plus3it/gravitybee/issues/363

**Comment by miurahr** [comment]
@nikitalita  Do you have any idea for that?

**Comment by nikitalita** [comment]
I've spent a couple hours trying to debug; the root cause is that script_path is no longer being found.


This is because `aqt-script.py` is no longer being created when we install via `pip install -e .[test,docs,check]`
2.2.1 build:
```
GravityBee CLI, 0.4.0
2022-08-09 05:01:06,070 - gravitybee - INFO - Arguments:
2022-08-09 05:01:06,070 - gravitybee - INFO - app_name: aqtinstall
2022-08-09 05:01:06,070 - gravitybee - INFO - app_version: 2.2.1
2022-08-09 05:01:06,070 - gravitybee - INFO - operating_system: windows
2022-08-09 05:01:06,070 - gravitybee - INFO - machine_type: amd64
2022-08-09 05:01:06,070 - gravitybee - INFO - console_script: aqt
2022-08-09 05:01:06,070 - gravitybee - INFO - pkg_name: aqt
2022-08-09 05:01:06,070 - gravitybee - INFO - script_path: D:\a\aqtinstall\aqtinstall\build_venv\Scripts\aqt-script.py
```
```
GravityBee CLI, 0.4.0
2022-08-12 14:14:30,503 - gravitybee - INFO - Arguments:
2022-08-12 14:14:30,503 - gravitybee - INFO - app_name: aqtinstall
2022-08-12 14:14:30,503 - gravitybee - INFO - app_version: 2.2.2
2022-08-12 14:14:30,503 - gravitybee - INFO - operating_system: windows
2022-08-12 14:14:30,503 - gravitybee - INFO - machine_type: amd64
2022-08-12 14:14:30,503 - gravitybee - INFO - console_script: aqt
2022-08-12 14:14:30,503 - gravitybee - INFO - pkg_name: aqt
2022-08-12 14:14:30,503 - gravitybee - INFO - script_path: None
```

The reason as near I can figure out: Pip's behavior changed at some point between aqt 2.2.1 and aqt 2.2.2, and pyproject.toml gets used to install instead of setup.cfg. Renaming pyproject.toml to something else and then running `pip install -e .[test,docs,check]` results in aqt-scripy.py being created.

I am not sure what resulted in this change; the package dependencies appear to be the same, setuptools is the same version (63.4.1), pip is the same version (22.2.2), etc. Testing this locally, trying with aqt 2.2.1-2.1.0 result in the same error.

So, I think just adding `move pyproject.toml pyproject.toml.bak` to build-standalone.bat is going to be the best we can do for now, and make sure we keep setup.cfg and pyproject.toml in sync moving forward.

**Comment by ddalcino** [comment]
> I am not sure what resulted in this change; the package dependencies appear to be the same

FWIW, the dependencies changed between aqt 2.2.1 and 2.2.2: In #547, I added the library `humanize` to `pyproject.toml` but I forgot to add it to `setup.cfg`. I thought that adding `humanize` to `setup.cfg` might fix the issue (#554), but [this run](https://github.com/ddalcino/aqtinstall/runs/7820851023?check_suite_focus=true) shows that it had no effect.

**Comment by nikitalita** [comment]
I messed with the configuration a lot; it's not anything to do with the added dependencies. Again, I attempted this locally with AQT 2.2.1, AQT 2.1, etc.; 

It's nothing to do with AQT, something happened with pip and or setuptools between the release of 2.2.1 and 2.2.2. I'm not sure what, the versions installed in the venv are the same in both the 2.2.1 and 2.2.2 runs, and I made sure I installed those versions when testing. I think it is likely that it could take some deep digging in the deps for pip and setuptools to figure out what happened here.
